heka/sandbox/decoders/extract_telemetry_dimensions.lua

```diff
@@ -143,21 +143,32 @@ function process_message()
         msg.Fields.clientId         = parsed.clientID
     elseif parsed.version then
         -- New-style telemetry, see http://mzl.la/1zobT1S
-        if parsed.type == "main" then
-            split_objects(parsed.environment, "environment", environment_objects)
-            split_objects(parsed.payload, "payload", main_ping_objects)
-            local ok, json = pcall(cjson.encode, parsed) -- re-encode the remaining data
-            if not ok then return -1, json end
-            msg.Payload = json
-        else
-            msg.Payload = json
+
+        -- pull out/verify the data/schema before any restructuring
+        local app = parsed.application
+        if type(app) ~= "table" then
+            return -1, "missing application object"
+        end
+
+        if type(parsed.payload.info) ~= "table" then
+           return -1, "missing info object"
```

Comment:
Right, I don't necessarily want to force all future ping types to include an `info` object if they don't have a `reason` field.